File: src/main/java/org/ecews/renaissance/domain/DentalConsultation.java
/*
* MIT License
*
* Copyright (c) 2025 [AUTHOR OR ORGANIZATION]
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in
* all copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
* THE SOFTWARE.
*/
package org.ecews.renaissance.domain;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@Entity
@Table(name = "dental_consultation")
public class DentalConsultation extends AuditEntity {
    
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;
    
    @ManyToOne(optional = false)
    @JoinColumn(name = "patient_id")
    private Patient patient;
    
    @Column(name = "diagnoses", columnDefinition = "text[]")
    private List<String> diagnoses;
    
    @Column(name = "treatments", columnDefinition = "text[]")
    private List<String> treatments;
    
    @Column(name = "dispensed_items", columnDefinition = "text[]")
    private List<String> dispensedItems;
    
    @Column(name = "services_referred", columnDefinition = "text[]")
    private List<String> servicesReferred;
    
    private Boolean referred;
    
    private List<String> othersDiagnosis;
    private List<String> othersTreatment;
}

File: Renaissance.AspNet/backend/Renaissance.Domain/Entities/DentalConsultation.cs
using System.Text.Json.Serialization;

namespace Renaissance.Domain.Entities;

public class DentalConsultation : AuditEntity
{
    public Guid Id { get; set; }
    public Guid PatientId { get; set; }

    [JsonIgnore]
    public Patient? Patient { get; set; }

    public List<string> Diagnoses { get; set; } = new List<string>();
    public List<string> Treatments { get; set; } = new List<string>();
    public List<string> DispensedItems { get; set; } = new List<string>();
    public List<string> ServicesReferred { get; set; } = new List<string>();
    public List<string> OthersDiagnosis { get; set; } = new List<string>();
    public List<string> OthersTreatment { get; set; } = new List<string>();
    public bool? Referred { get; set; }
}
